### ISSUE TITLE
Absolute path for OpenSLL libraries in PocoCryptoTargets.cmake

### ISSUE THREAD

**Issue by andioz** [pocoproject/poco#906]
I build POCO 1.6.0 or POCO 1.6.1 using CMake. I compile including NetSSL (using prebuilt binaries for OpenSSL). Then I distribute the binaries in an "external" SVN repository to be used by other projects. These other projects uses find_package(), PocoConfig.cmake etc. System is Windows 7 Win32/x64 builds. 

Now i saw that in some files in folder lib\cmake\Poco (those which depends on OpenSSL of course) the absolute path to the OpenSSL libs is used. For example lib\cmake\Poco\PocoCryptoTargets.cmake:

```
set_target_properties(Poco::Crypto PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Poco::Foundation;$<$<NOT:$<CONFIG:DEBUG>>:D:/Download/OpenSSL/OpenSSL-Win32/lib/VC/ssleay32MD.lib>;$<$<CONFIG:DEBUG>:D:/Download/OpenSSL/OpenSSL-Win32/lib/VC/ssleay32MDd.lib>;$<$<NOT:$<CONFIG:DEBUG>>:D:/Download/OpenSSL/OpenSSL-Win32/lib/VC/libeay32MD.lib>;$<$<CONFIG:DEBUG>:D:/Download/OpenSSL/OpenSSL-Win32/lib/VC/libeay32MDd.lib>"
)
```

This breaks builds on other machines (VS2010), because the OpenSSL libraries are included from a differnet directory. I wonder, can this absolute path's be avoided?

**Comment by bachp** [comment]
The file you mentioned: `lib\cmake\Poco\PocoCryptoTargets.cmake` is generated by cmake at build time. So it will contain the library used at that time.

I'm not sure how this would work on windows with non absolute paths as there is default location where to search for the OpenSSL libraries. Please correct me if I am wrong.

**Comment by andioz** [comment]
My current workaround is as follows: I parse the list in INTERFACE_LINK_LIBRARIES and remove every enty for OpenSLL, or in other words, every reference to an "external" library. This way the variable INTERFACE_LINK_LIBRARIES contains only dependencies to internal targets, which are "logical" targets, not real files. 

This is fine for me. I think it is not necessary to add all dependencies to external libraries. Let the user do this in his CMakeLists.txt using find_package(). What do you think?

**Comment by agalanin** [comment]
The same problem with Poco 1.9.0 on both UNIX and Windows platforms.

Work-around:

```CMake
set_target_properties(Poco::Crypto PROPERTIES
    INTERFACE_LINK_LIBRARIES Poco::Foundation OpenSSL::SSL OpenSSL::Crypto
)
set_target_properties(Poco::NetSSL PROPERTIES
    INTERFACE_LINK_LIBRARIES Poco::Crypto Poco::Util Poco::Net
)
```

**Comment by bachp** [comment]
I think the problem is that Poco is not using the importet targets from `FindOpenSSL.cmake` right now.
If this would be change the issue might be gone. But for this to work the minimal supported CMake version needs to be increased to 3.4.

@agalanin What CMake version are you using?

**Comment by agalanin** [comment]
> @agalanin What CMake version are you using?

3.11

**Comment by Bjoe** [comment]
@bachp Right!

This issue is fixed in poco version 1.9.1. In 1.9.1 all third party libraries are included as import targets. It also works with cmake 3.3, because I add a workaround for that.

**Comment by agalanin** [comment]
> This issue is fixed in poco version 1.9.1. In 1.9.1 all third party libraries are included as import targets. It also works with cmake 3.3, because I add a workaround for that.

I confirm that this issue is fixed somewhere between 1.9.0 and 1.9.1:

--- TEST-1.9.0/usr/local/lib/cmake/Poco/PocoCryptoTargets.cmake 2018-11-27 22:31:38.000000000 +0300
+++ TEST-1.9.1/usr/local/lib/cmake/Poco/PocoCryptoTargets.cmake 2018-11-27 21:59:28.000000000 +0300
@@ -54,9 +54,8 @@
 add_library(Poco::Crypto SHARED IMPORTED)
 ```diff
 set_target_properties(Poco::Crypto PROPERTIES
-  INTERFACE_COMPILE_DEFINITIONS "POCO_NO_AUTOMATIC_LIBS"
   INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
-  INTERFACE_LINK_LIBRARIES "Poco::Foundation;/usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so"
+  INTERFACE_LINK_LIBRARIES "Poco::Foundation;OpenSSL::SSL;OpenSSL::Crypto"
 )
 
 if(CMAKE_VERSION VERSION_LESS 2.8.12)
```

**Comment by Bjoe** [comment]
@agalanin Thanks for the verification!
